### PR TITLE
Allow to store files in sub-folder

### PR DESCRIPTION
**Pull request by jeremyallebe**
This pull request allow to store files via `addFile()` function into a sub folder of the archive.

### CODE REVIEW

File: src/PersonalDataSelection.php [hunk 1]
```diff
@@ -48,15 +48,21 @@ public function add(string $nameInDownload, array | string $content): PersonalDa
         return $this;
     }
 
-    public function addFile(string $pathToFile, string $diskName = null)
+    public function addFile(string $pathToFile, string $diskName = null, string $folder = null)
```
Comment (freekmurze) [inline src/PersonalDataSelection.php:51]: Rename `folder` to `directory`

File: src/PersonalDataSelection.php [hunk 2]
```diff
@@ -48,15 +48,21 @@ public function add(string $nameInDownload, array | string $content): PersonalDa
...
-            ? $this->copyLocalFile($pathToFile)
-            : $this->copyFileFromDisk($pathToFile, $diskName);
+            ? $this->copyLocalFile($pathToFile, $folder)
+            : $this->copyFileFromDisk($pathToFile, $diskName, $folder);
     }
 
-    protected function copyLocalFile(string $pathToFile)
+    protected function copyLocalFile(string $pathToFile, string $folder = null)
     {
+        if (is_string($folder) && !empty($folder)) {
```
Comment (freekmurze) [inline src/PersonalDataSelection.php:60]: This is unreadable to me. Extract this to a function with a good name

File: src/PersonalDataSelection.php [hunk 3]
```diff
@@ -48,15 +48,21 @@ public function add(string $nameInDownload, array | string $content): PersonalDa
...
+            $fileName = $folder . '/' . pathinfo($pathToFile, PATHINFO_BASENAME);
+        } else {
```
Comment (freekmurze) [inline src/PersonalDataSelection.php:62]: I'm not a fan of `else` could you refactor that away (possibly by using a ternary?

**Review by freekmurze — CHANGES_REQUESTED** [review]
Could you add tests that prove that this works and also update the readme?

File: src/PersonalDataSelection.php (hunk 1 above)
Comment (jeremyallebe) [reply src/PersonalDataSelection.php]: I renamed it

File: src/PersonalDataSelection.php (hunk 2 above)
Comment (jeremyallebe) [reply src/PersonalDataSelection.php]: I refactored it

File: src/PersonalDataSelection.php (hunk 3 above)
Comment (jeremyallebe) [reply src/PersonalDataSelection.php]: I refactored it

**Comment by jeremyallebe** [comment]
I refactored some part of my code and add some tests.

**Comment by jeremyallebe** [comment]
Yes sure, i updated it

File: README.md [hunk 4]
```diff
@@ -37,6 +37,15 @@ public function selectPersonalData(PersonalDataSelection $personalDataSelection)
 }
 ```
 
+You can store files in a directory of the archive. For this, add the directory path as the third parameter.
+
+```php
+public function selectPersonalData(PersonalDataSelection $personalDataSelection) {
+    $personalDataSelection
+        ->addFile(storage_path("avatars/{$this->id}.jpg"), null, 'avatars');
```
Comment (freekmurze) [inline README.md:45]: Could you use a named parameter here, so we can avoid `null` in this example?

**Review by freekmurze — CHANGES_REQUESTED** [review]

**Comment by jeremyallebe** [comment]
Hey @freekmurze,
can you merge the pull request? Or maybe some other changes to do?
Thanks

**Comment by freekmurze** [comment]
Thank you!

**Pull request merged**